daml-lf/scenario-interpreter/src/perf/benches/scala/com/digitalasset/daml/lf/speedy/perf/CollectAuthority.scala

```diff
@@ -15,27 +15,37 @@ import java.io.File
 import java.util.concurrent.TimeUnit
 import org.openjdk.jmh.annotations._
 
-@State(Scope.Thread)
+@State(Scope.Benchmark)
 class CollectAuthorityState {
-  private val darFile = new File(rlocation("daml-lf/scenario-interpreter/CollectAuthority.dar"))
-  private val packages = UniversalArchiveReader().readFile(darFile).get
-  private val packagesMap = Map(packages.all.map {
-    case (pkgId, pkgArchive) => Decode.readArchivePayloadAndVersion(pkgId, pkgArchive)._1
-  }: _*)
-
-  // NOTE(MH): We use a static seed to get reproducible runs.
-  private val seeding = crypto.Hash.secureRandom(crypto.Hash.hashPrivateKey("scenario-perf"))
-  private val buildMachine = Speedy.Machine
-    .newBuilder(
-      PureCompiledPackages(packagesMap).right.get,
-      Time.Timestamp.MinValue,
-      Some(seeding()))
-    .fold(err => sys.error(err.toString), identity)
-  private val expr = EVal(
-    Identifier(packages.main._1, QualifiedName.assertFromString("CollectAuthority:test")))
-  // NOTE(MH): We run the machine once to initialize all data that is shared
-  // between runs.
-  run()
+  private var buildMachine: Expr => Speedy.Machine = _
+  private var expr: Expr = _
```

Comment:
I don’t really like the underscore stuff. Just setting the default explicitly is easier to read imho.

Reply:
I don't see an easy way to avoid those underscores without duplicating the body of the `init` method here. DO you have an idea?

Reply:
the underscore will just set it to `null` so just set it to `null` explicitly, i.e., `private var expr: Expr = null`.

Reply:
Oh right, there was this weird `null` thingy...